src/cryptography/hazmat/primitives/poly1305.py

```diff
@@ -2,16 +2,20 @@
 # 2.0, and the BSD License. See the LICENSE file in the root of this repository
 # for complete details.
 
+import typing
 
 from cryptography import utils
 from cryptography.exceptions import (
     AlreadyFinalized,
     UnsupportedAlgorithm,
     _Reasons,
 )
+from cryptography.hazmat.backends.openssl.poly1305 import _Poly1305Context
```

Comment:
We could potentially remove this import if we changed our model of finalization to not check whether `_ctx` is `None` or not.